src/hotspot/share/opto/type.cpp

```diff
@@ -3223,14 +3223,22 @@ const TypePtr* TypeRawPtr::add_offset(intptr_t offset) const {
   case TypePtr::NotNull:
     return this;
   case TypePtr::Null:
+    return make( (address)offset );
```

Comment:
Shouldn't this assert that _bits == 0?  Looking at the code, however, I can't find anywhere that we actually create a TypeRawPtr with TypePtr::Null.  We could probably remove this case and let it fall through to the default ShouldNotReachHere().

Reply:
Initialization of `TypePtr::NULL_PTR` here:
https://github.com/openjdk/jdk/blob/4d50cbb5a73ad1f84ecd6a895045ecfdb0835adc/src/hotspot/share/opto/type.cpp#L538

Reply:
I saw that too, but it creates a TypePtr, not a TypeRawPtr.

Reply:
Oh, you are right.  And TypeRawPtr::make asserts the PTR is neither Constant nor Null.  Which makes
both switch cases under modification here supposedly unreachable.  That would explain why I never hit
either after running lots of tests.  All of the change proposed here can be eliminated, and instead change
both cases to fall through to the default ShouldNotReachHere().  (And that would be another way to
remove the `-Wzero-as-null-pointer-constant` warning that was how I got here in the first place. :) )

Reply:
Oh, you are right. And TypeRawPtr::make asserts the PTR is neither Constant nor Null. Which makes
both switch cases under modification here supposedly unreachable. That would explain why I never hit
either after running lots of tests. All of the change proposed here can be eliminated, and instead change
both cases to fall through to the default ShouldNotReachHere(). (And that would be another way to
remove the -Wzero-as-null-pointer-constant warning that was how I got here in the first place. :) )

Reply:
There's TypeRawPtr::make(enum PTR ptr) which doesn't allow Constant or Null, but we are using TypeRawPtr::make(address bits) here.
We may need to keep the Constant case.  I wouldn't be surprised if there was a way to trigger that path using Unsafe.

Reply:
Yeah, keeping it makes sense.  I've removed the TypePtr::Null case, allowing that one to default to
ShuoldNotReachHere().